### ISSUE TITLE
Macros panic when the input length exceeds 27 characters

### ISSUE THREAD

**Issue by golddranks** [dtolnay/proc-macro-hack#5]
`rustc 1.15.1 (021bd294c 2017-02-08)`.
For some reason, proc macros panic when the input length exceeds 27 characters:
```
error: custom derive attribute panicked
  --> tests/tests.rs:13:5
   |
13 |     assert_eq!(base2!("01100001011000010110000101100001"), b"aaaa");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: message: `base2_impl` procedural macro failed
   = note: this error originates in a macro outside of the current crate

error: Could not compile `binary_macros`.
```
Versus: `assert_eq!(base2!("011000010110000101100001"), b"aaa");` which compiles and runs without problems.

I'm experiencing this with all of my macros, even trivial ones like this:
```
    pub fn trivial_test_impl(input: &str) -> String {
        input.to_owned()
    }
```
and
```
proc_macro_expr_decl!(trivial_test! => trivial_test_impl);
```
and
```
#[test]
fn trivial_test_fails() {
    assert_eq!(trivial_test!("abcdefghijklmnopqrstuvwxyz"), "abcdefghijklmnopqrstuvwxyz");
}
```
This fails, but runs if I take one letter away.

**Comment by golddranks** [comment]
Btw. if you want to have a quick test case demonstrating this, check out the latest commit on master of this crate: https://github.com/golddranks/binary_macros/commit/92b39dd6d4a4040af7a13f8b29525f343fd0882b

**Comment by dtolnay** [comment]
Thanks! I fixed this in proc-macro-hack 0.3.1 which means people using binary_macros 0.5.0 will receive the fix immediately.
